samza-core/src/test/java/org/apache/samza/zk/TestZkSystemEnvironmentSetter.java

```diff
@@ -0,0 +1,116 @@
+
+import static org.junit.Assert.*;
+
+
+public class TestZkSystemEnvironmentSetter {
+
+  private byte[] originalProperties;
+
+  @Before
+  public void before() throws Throwable {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    System.getProperties().store(baos, "");
+    baos.close();
+    originalProperties = baos.toByteArray();
+  }
+
+  @After
+  public void after() {
+    try (ByteArrayInputStream bais = new ByteArrayInputStream(originalProperties)) {
+      System.getProperties().clear();
+      System.getProperties().load(bais);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testConfig() {
+    Map<String, String> configMap = new HashMap<>();
+    configMap.put("samza.system.zookeeper.client.secure", "true");
+    configMap.put("samza.system.zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty");
+    configMap.put("samza.system.zookeeper.ssl.keyStore.location", "keyStoreLocation");
+    configMap.put("samza.system.zookeeper.ssl.keyStore.password", "keyStorePassword");
+    configMap.put("samza.system.zookeeper.ssl.keyStore.type", "PKCS12");
+    configMap.put("samza.system.zookeeper.ssl.trustStore.location", "trustStoreLocation");
+    configMap.put("samza.system.zookeeper.ssl.trustStore.password", "trustStorePassword");
+    configMap.put("samza.system.zookeeper.ssl.trustStore.type", "JKS");
+    Config config = new MapConfig(configMap);
+
+    ZkSystemEnvironmentSetter.setZkEnvironment(config);
+    assertEquals(System.getProperty("zookeeper.client.secure"), "true");
+    assertEquals(System.getProperty("zookeeper.clientCnxnSocket"), "org.apache.zookeeper.ClientCnxnSocketNetty");
+    assertEquals(System.getProperty("zookeeper.ssl.keyStore.location"), "keyStoreLocation");
+    assertEquals(System.getProperty("zookeeper.ssl.keyStore.password"), "keyStorePassword");
+    assertEquals(System.getProperty("zookeeper.ssl.keyStore.type"), "PKCS12");
+    assertEquals(System.getProperty("zookeeper.ssl.trustStore.location"), "trustStoreLocation");
+    assertEquals(System.getProperty("zookeeper.ssl.trustStore.password"), "trustStorePassword");
+    assertEquals(System.getProperty("zookeeper.ssl.trustStore.type"), "JKS");
+
+    System.clearProperty("zookeeper.client.secure");
+    System.clearProperty("zookeeper.clientCnxnSocket");
+    System.clearProperty("zookeeper.ssl.keyStore.location");
+    System.clearProperty("zookeeper.ssl.keyStore.password");
+    System.clearProperty("zookeeper.ssl.keyStore.type");
+    System.clearProperty("zookeeper.ssl.trustStore.location");
+    System.clearProperty("zookeeper.ssl.trustStore.password");
+    System.clearProperty("zookeeper.ssl.trustStore.type");
+  }
+
+  @Test
+  public void testEmptyConfig() {
+    Config config = new MapConfig(Collections.emptyMap());
+
+    ZkSystemEnvironmentSetter.setZkEnvironment(config);
+    assertEquals(System.getProperty("zookeeper.client.secure"), null);
```

Comment:
are the accessor of these properties null safe?

Reply:
yes. Then properties are null by default.